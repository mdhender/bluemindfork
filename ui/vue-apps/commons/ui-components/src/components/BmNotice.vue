<template>
    <div class="bm-notice d-flex px-2">
        <div class="arrow-up" />
        <span class="text-center w-100 text-condensed py-1">{{ text }}</span>
    </div>
</template>

<script>
export default {
    name: "BmNotice",
    props: {
        text: { type: String, default: "" }
    }
};
</script>
<style lang="scss">
@import "../css/_variables.scss";

.bm-notice {
    margin: $tooltip-arrow-height 0 0;
    background-color: $fill-danger-bg;

    .arrow-up {
        position: relative;
        display: block;
        width: $tooltip-arrow-width;
        height: $tooltip-arrow-height;
        bottom: $tooltip-arrow-height;
        left: 50%;

        &::before {
            position: absolute;
            content: "";
            border-color: transparent;
            border-style: solid;
            bottom: 0;
            border-width: 0 calc($tooltip-arrow-width / 2) $tooltip-arrow-height;
            border-bottom-color: $danger-fg;
        }
    }

    span {
        color: $fill-danger-fg;
    }
}
</style>
