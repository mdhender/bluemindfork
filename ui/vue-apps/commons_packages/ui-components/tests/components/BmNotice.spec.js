import { mount } from "@vue/test-utils";
import BmNotice from "../../src/components/BmNotice";

describe("BmNotice", () => {
    function defaultMount() {
        return mount(BmNotice);
    }

    test("is a Vue instance", () => {
        expect(defaultMount().vm).toBeTruthy();
    });

    test("should match snapshot", () => {
        expect(defaultMount().vm.$el).toMatchSnapshot();
    });

    test("should have a bm-alert class", () => {
        expect(defaultMount().vm.$el.className).toEqual(expect.stringContaining("bm-notice"));
    });
});
