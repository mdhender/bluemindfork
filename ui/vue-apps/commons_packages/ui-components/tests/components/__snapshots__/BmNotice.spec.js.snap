// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BmNotice should match snapshot 1`] = `
<div
  class="bm-notice d-flex px-2"
>
  <div
    class="arrow-up"
  />
   
  <span
    class="text-center w-100 text-condensed py-1"
  >
    
  </span>
</div>
`;
